openmls/src/group/mls_group/mod.rs

```diff
@@ -72,50 +72,107 @@ implement_persistence!(
     mls_version
 );
 
-/// Public `MlsGroup` functions.
-impl MlsGroup {
-    pub fn new(
-        id: &[u8],
-        ciphersuite_name: CiphersuiteName,
-        backend: &impl OpenMlsCryptoProvider,
-        key_package_bundle: KeyPackageBundle,
-        config: MlsGroupConfig,
-        psk_option: impl Into<Option<PskSecret>>,
-        version: impl Into<Option<ProtocolVersion>>,
-    ) -> Result<Self, MlsGroupError> {
-        debug!("Created group {:x?}", id);
-        trace!(" >>> with {:?}, {:?}", ciphersuite_name, config);
-        let group_id = GroupId { value: id.into() };
-        let ciphersuite = Config::ciphersuite(ciphersuite_name)?;
-        let tree = RatchetTree::new(backend, key_package_bundle);
-        // TODO #483: Implement extensions
-        let extensions: Vec<Extension> = Vec::new();
+/// Builder for [`MlsGroup`].
+pub struct MlsGroupBuilder {
+    key_package_bundle: KeyPackageBundle,
+    group_id: Option<GroupId>,
+    config: Option<MlsGroupConfig>,
+    psk: Option<PskSecret>,
+    version: Option<ProtocolVersion>,
+    required_capabilities: Option<RequiredCapabilitiesExtension>,
+}
+
+impl MlsGroupBuilder {
+    /// Create a new [`MlsGroupBuilder`].
+    pub fn new(key_package_bundle: KeyPackageBundle) -> Self {
+        Self {
+            key_package_bundle,
+            group_id: None,
+            config: None,
+            psk: None,
+            version: None,
+            required_capabilities: None,
+        }
+    }
+    /// Set the [`GroupId`] of the [`MlsGroup`].
+    pub fn with_group_id(mut self, group_id: GroupId) -> Self {
+        self.group_id = Some(group_id);
+        self
+    }
+    /// Set the [`GroupId`] of the [`MlsGroup`].
+    pub fn with_group_id_slice(mut self, group_id: &[u8]) -> Self {
```

Comment:
Is this really needed? We can always build a `GroupId` from a slice.

Reply:
We sometimes have a `GroupId` and sometimes a slice. Having both is more flexible and doesn't hurt.